### PR TITLE
fix: Resend OTP button

### LINKED ISSUE #435: Resend OTP button disappears when you enter an invalid otp
**Describe the bug**
Resend OTP button disappears when you enter an invalid otp

**To Reproduce**
- Send an OTP
- Wait for Resend OTP to appear
- Enter an invalid otp
- Watch the Resend OTP button disappear

**Expected behavior**
- Send an OTP
- Wait for Resend OTP to appear
- Enter an invalid otp
- Able to click on resend otp to get a new otp

### PR DESCRIPTION
**Pull request by lamkeewei**
## Problem

Resend OTP button disappears after user tries to login with an invalid OTP.

Closes #435 

## Solution

**Improvements**:

- Display more relevant error message returned by the API when retrieving OTP. E.g. "Wait for X seconds before requesting for a new otp" 

**Bug Fixes**:

- Do not reset `canResend` on login to `false`. 
- Do not reset `otpSent` to `false` on resend. This was previously causing the input field to jump back to the email field briefly before going back to the OTP field after the API request is sent. 

## Before & After Screenshots

**AFTER**:
image

## Tests

**Disappearing resend button**
1. Request for an OTP
2. Wait for Resend OTP to be enabled
3. Enter an invalid OTP
4. Resend OTP should still be enabled

**Get OTP error messages**
1. Request for an OTP
2. Refresh page immediately and request for an OTP again
3. Error message should show "Wait for X seconds before requesting for a new otp" instead.

### CODE REVIEW

**Comment by jeantanzj** [comment]
Screenshot 2020-06-25 at 3 44 01 PM
Error message is too detailed

See the chapter on Verbose Failure Messages in The Web Application Hackers handbook

File: frontend/src/components/login/login-input/LoginInput.tsx [hunk 1]
```diff
@@ -69,12 +72,26 @@ const Login = () => {
...
+    } catch (err) {
+      setErrorMsg(err.message)
+      sendException(err.message)
+    } finally {
+      setIsResending(false)
+      // Always reset timer even if resending fails
+      setTimeout(() => {
+        setCanResend(true)
+      }, RESEND_WAIT_TIME)
+    }
```
Comment (pregnantboy) [inline frontend/src/components/login/login-input/LoginInput.tsx:94]: why not
```
    setIsResending(true)
    await sendOtp()
    setIsResending(false)
```

Comment (pregnantboy) [reply frontend/src/components/login/login-input/LoginInput.tsx:94]: and in catch block of `sendOtp()`
```
   setCanResend(true)
```

Comment (lamkeewei) [reply frontend/src/components/login/login-input/LoginInput.tsx:94]: I was thinking of enforcing the timeout even in the event that the resend failed (network error, etc). Not sure if that's the required behaviour though.

**Pull request merged**